tests/unit/s2n_server_supported_versions_extension_test.c

```diff
@@ -0,0 +1,178 @@
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+
+#include <stdint.h>
+
+#include "tls/s2n_alerts.h"
+#include "tls/s2n_config.h"
+#include "tls/s2n_connection.h"
+#include "tls/s2n_tls.h"
+#include "tls/extensions/s2n_server_supported_versions.h"
+
+#include "stuffer/s2n_stuffer.h"
+#include "utils/s2n_safety.h"
+
+#define ILLEGAL_PARAMETER_ALERT 47
+
+int get_alert(struct s2n_connection *conn) {
+    uint8_t error[2];
+    GUARD(s2n_stuffer_read_bytes(&conn->reader_alert_out, error, 2));
+    return error[1];
+}
+
+int write_test_supported_version(struct s2n_stuffer *list, uint8_t supported_version) {
+    GUARD(s2n_stuffer_write_uint8(list, S2N_TLS_PROTOCOL_VERSION_LEN));
+
+    GUARD(s2n_stuffer_write_uint8(list, supported_version / 10));
+    GUARD(s2n_stuffer_write_uint8(list, supported_version % 10));
+
+    return 0;
+}
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+
+    EXPECT_SUCCESS(setenv("S2N_ENABLE_TLS13_FOR_TESTING", "1", 0));
+    uint8_t latest_version = S2N_TLS13;
+
+    struct s2n_config *config;
+    EXPECT_NOT_NULL(config = s2n_config_new());
+
+    /* Server sends a supported_version the client can parse */
+    {
+    	struct s2n_connection *server_conn;
+    	EXPECT_NOT_NULL(server_conn = s2n_connection_new(S2N_SERVER));
+    	EXPECT_SUCCESS(s2n_connection_set_config(server_conn, config));
+
+    	uint16_t expected_length = 6;
+
+    	struct s2n_stuffer extension;
+    	s2n_stuffer_alloc(&extension, expected_length);
+
+    	EXPECT_SUCCESS(s2n_extensions_server_supported_versions_send(server_conn, &extension));
+
+    	/* Check that type and size are correct */
+    	uint16_t extension_type;
+    	s2n_stuffer_read_uint16(&extension, &extension_type);
+    	EXPECT_EQUAL(extension_type, TLS_EXTENSION_SUPPORTED_VERSIONS);
+    	uint16_t extension_length;
+    	s2n_stuffer_read_uint16(&extension, &extension_length);
+    	EXPECT_EQUAL(extension_length, s2n_stuffer_data_available(&extension));
+
+    	/* Check that the client can process the version */
+    	struct s2n_connection *client_conn;
+    	EXPECT_NOT_NULL(client_conn = s2n_connection_new(S2N_CLIENT));
+    	EXPECT_SUCCESS(s2n_connection_set_config(client_conn, config));
+
+    	EXPECT_SUCCESS(s2n_extensions_server_supported_versions_recv(client_conn, &extension));
+    	EXPECT_EQUAL(client_conn->client_protocol_version, latest_version);
+    	EXPECT_EQUAL(client_conn->server_protocol_version, latest_version);
+    	EXPECT_EQUAL(client_conn->actual_protocol_version, latest_version);
+
+    	/* Clean up */
+    	EXPECT_SUCCESS(s2n_connection_free(client_conn));
+    	EXPECT_SUCCESS(s2n_connection_free(server_conn));
+    	EXPECT_SUCCESS(s2n_stuffer_free(&extension));
+
```

Comment:
/nit there's usually no trailing new lines before closing brackets, similar for the subsequent blocks

Reply:
easy fix